std/datetime.d

```diff
@@ -310,84 +310,184 @@ public:
     /++
         Returns the current time in the given time zone.
 
+        Params:
+            clockType = The $(CXREF time, ClockType) indicates which system
+                        clock to use to get the current time. Very few programs
+                        need to use anything other than the default.
+            tz = The time zone for the SysTime that's returned.
+
         Throws:
-            $(XREF exception, ErrnoException) (on Posix) or $(XREF exception, Exception) (on Windows)
-            if it fails to get the time of day.
+            $(LREF DateTimeException) if it fails to get the time.
       +/
-    static SysTime currTime(immutable TimeZone tz = LocalTime()) @safe
+    static SysTime currTime(ClockType clockType = ClockType.normal)(immutable TimeZone tz = LocalTime()) @safe
     {
-        return SysTime(currStdTime, tz);
+        return SysTime(currStdTime!clockType, tz);
     }
 
     unittest
     {
+        import std.format : format;
+        assert(currTime().timezone is LocalTime());
         assert(currTime(UTC()).timezone is UTC());
 
-        //I have no idea why, but for some reason, Windows/Wine likes to get
-        //time_t wrong when getting it with core.stdc.time.time. On one box
-        //I have (which has its local time set to UTC), it always gives time_t
-        //in the real local time (America/Los_Angeles), and after the most recent
-        //DST switch, every Windows box that I've tried it in is reporting
-        //time_t as being 1 hour off of where it's supposed to be. So, I really
-        //don't know what the deal is, but given what I'm seeing, I don't trust
-        //core.stdc.time.time on Windows, so I'm just going to disable this test
-        //on Windows.
+        // core.stdc.time.time does not always work correctly on Windows systems.
+        // In particular, sometimes it applies the local DST to time_t, even
+        // though time_t is in UTC. I'm fairly certain that it's a bug in dmc's
+        // implementation, but it needs to be investigated. Regardless, the
+        // result is that for now, we can't run this test on Windows.
         version(Posix)
         {
+            static import std.math;
             immutable unixTimeD = currTime().toUnixTime();
             immutable unixTimeC = core.stdc.time.time(null);
-            immutable diff = unixTimeC - unixTimeD;
-
-            assert(diff >= -2);
-            assert(diff <= 2);
+            assert(std.math.abs(unixTimeC - unixTimeD) <= 2);
         }
+
+        auto norm1 = Clock.currTime;
```

Comment:
I think I've seen this exact code elsewhere. I believe that unittests are as good place for using a bit of DRY as anything else.